src/fuse_reduce.cpp

```diff
@@ -197,13 +197,31 @@ struct find_pointwise_reduce
         return match::name("fused_reduce")(match_broadcastable_input("pointwise", "pointwise"));
     }
 
+    bool is_valid_broadcast(const instruction_ref b, const std::vector<size_t> reduce_axes) const
+    {
+        std::vector<size_t> broadcast_indices;
+        auto bstrides = b->get_shape().strides();
+
+        return std::all_of(
+            reduce_axes.begin(), reduce_axes.end(), [&](auto a) { return bstrides.at(a) == 0; });
+    }
+
     void apply(module_pass_manager& mpm, const match::matcher_result& r) const
     {
         auto reduce        = r.result;
         auto input         = r.instructions["pointwise"];
         const auto* pm     = input->module_inputs().front();
         const auto* old_rm = reduce->module_inputs().front();
 
+        if(contains(r.instructions, "broadcast"))
+        {
+            auto axes       = reduce->get_operator().to_value().at("axes").to_vector<size_t>();
+            auto broadcast  = r.instructions["broadcast"];
+            auto fbroadcast = r.instructions["final_broadcast"];
+            if(not(is_valid_broadcast(broadcast, axes) and is_valid_broadcast(fbroadcast, axes)))
```

Comment:
You should just check "broadcast" and not "final_broadcast"(which is the broadcast after contiguous if there is one).